docs/source/eventproc.rst

```diff
@@ -164,27 +164,74 @@ The remove an existing event handler, a `deregister` function is available.  It
     >>> deregister(conn, 5)
     True
 
-Uploading a File From a Handler
+
+Writing Handlers
+--------------------------
+
+Code executing in a handler can import and use packages and read/write files, similarly to how the handler function works in the environment you defined/registered the handler in.
+
+Using Packages
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-To save information from a handler into a file for later access, the `upload_file` function is available. It uploads file to S3 and returns a link to download the file.
-Below is an example of it's use. Further information about the function behavior can be found with `help(upload_file)`.
+
+Any packages used within a handler must be imported within that handler as follows.
```

Comment:
Is this true?  I remember we did work on this but can't remember the outcomes.  We at one point thought that if we "register" a function, then it would end up including any imports that function made use of.

Reply:
Perhaps we should say that a future version will allow users to "specify pip installable packages they want to use" or some such.  No big deal though... there are positives for not promising this to them.

Reply:
yes it's true